### PR TITLE
8269878: Handle redundant reg-2-reg moves in X86 backend

### PR DESCRIPTION
**Pull request by jatin-bhateja**
Chain patterns which connects various legacy, VL and non-legacy USE - DEF operands during selection need not emit a reg-2-reg move instruction. C2 register allocator can handle these cases by back propagating the register mask from USE to DEF e.g.

USE (regD) <- DEF( legRegD) : Not an issue since regD can accommodate ranges allocated to legRegD.

USE (legRegD) <- USE (regD) : Not an issue since Allocator will back propagate legRegD mask from USE to its DEF node, which will constrain the allocation set to lower register bank.

Same logic applies to vlRegD , rRegD, legRegD combinations.

These machine nodes are redundant post selection and can be removed during generic operand resolution phase.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269878](https://bugs.openjdk.java.net/browse/JDK-8269878): Handle redundant reg-2-reg moves in X86 backend

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to 62275736ad6992fbd9fca34704a6d6b6423a4d72
 * [Vladimir Ivanov](https://openjdk.java.net/census#vlivanov) (@iwanowww - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4693/head:pull/4693` \
`$ git checkout pull/4693`

Update a local copy of the PR: \
`$ git checkout pull/4693` \
`$ git pull https://git.openjdk.java.net/jdk pull/4693/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4693`

View PR using the GUI difftool: \
`$ git pr show -t 4693`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4693.diff">https://git.openjdk.java.net/jdk/pull/4693.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/x86/x86_32.ad [hunk 1]
```diff
@@ -5021,6 +5021,89 @@ define %{
 //               name must have been defined in an 'enc_class' specification
 //               in the encode section of the architecture description.
 
+// Chain instructions needed by selection algorithm.
```
Comment (vnkozlov) [inline src/hotspot/cpu/x86/x86_32.ad:5024]: Please update comment similar to `MoveVec2Leg` instruction:
```
// Dummy reg-to-reg vector moves. Removed during post-selection cleanup.
```

File: src/hotspot/cpu/x86/x86_64.ad [hunk 2]
```diff
@@ -4820,6 +4820,87 @@ define
 //               name must have been defined in an 'enc_class' specification
 //               in the encode section of the architecture description.
 
+// Chain instructions needed by selection algorithm.
```
Comment (vnkozlov) [inline src/hotspot/cpu/x86/x86_64.ad:4823]: Fix comment.

**Review by vnkozlov — COMMENTED** [review]
Other than comments it looks good to me.
I will test it.

**Review by vnkozlov — APPROVED** [review]
Testing tier1-3 passed.

**Review by iwanowww — APPROVED** [review]
Looks good.

**Comment by jatin-bhateja** [comment]
Thanks reviewes.

**Pull request closed without merging**